src/experimentation/handler.js

```diff
@@ -42,55 +28,40 @@ function processRUMResponse(data) {
       time95: row.time95,
     }));
 }
-export default async function auditExperiments(message, context) {
-  const { type, url: siteId, auditContext } = message;
-  const { dataAccess, log, sqs } = context;
-  const {
-    AUDIT_RESULTS_QUEUE_URL: queueUrl,
-  } = context.env;
-  try {
-    log.info(`Received audit request for siteId: ${siteId}`);
-    const site = await retrieveSiteBySiteId(dataAccess, siteId, log);
-    if (!site) {
-      return notFound('Site not found');
-    }
-    if (!site.isLive()) {
-      log.info(`Site ${siteId} is not live`);
-      return ok();
-    }
 
-    const rumAPIClient = RUMAPIClient.createFrom(context);
-    const finalUrl = await getRUMUrl(site.getBaseURL());
-    auditContext.finalUrl = finalUrl;
-    const params = {
-      url: finalUrl,
-    };
+async function processAudit(baseURL, context) {
+  const rumAPIClient = RUMAPIClient.createFrom(context);
+  const finalUrl = await getRUMUrl(baseURL);
+  const params = {
+    url: finalUrl,
+  };
+
+  const data = await rumAPIClient.getExperimentationData(params);
+  return {
+    auditResult: processRUMResponse(data),
+    fullAuditRef: createExperimentationURL({ url: finalUrl }),
+  };
+}
+
+export async function experimentationAuditRunner(baseURL, context) {
+  const { log } = context;
+  log.info(`Received Experimentation audit request for ${baseURL}`);
+  const startTime = process.hrtime();
 
-    const data = await rumAPIClient.getExperimentationData(params);
-    const auditResult = {
-      result: processRUMResponse(data),
-      finalUrl: auditContext.finalUrl,
-    };
+  const auditData = await processAudit(
+    baseURL,
+    context,
+  );
 
-    const auditData = {
-      siteId: site.getId(),
-      isLive: site.isLive(),
-      auditedAt: new Date().toISOString(),
-      auditType: type,
-      auditResult,
-      fullAuditRef: createExperimentationURL({ url: auditContext.finalUrl }),
-    };
-    await dataAccess.addAudit(auditData);
-    await sqs.sendMessage(queueUrl, {
-      url: site.getBaseURL(),
-      type,
-      auditContext,
-      auditResult,
-    });
-    log.info(`Successfully audited ${siteId} for ${type} type audit`);
-    return noContent();
-  } catch (e) {
-    log.error(`Audit ${type} failed for ${siteId}`, e);
-    return internalServerError(`Internal server error: ${e.message}`);
-  }
+  const endTime = process.hrtime(startTime);
+  const elapsedSeconds = endTime[0] + endTime[1] / 1e9;
+  const formattedElapsed = elapsedSeconds.toFixed(2);
+
+  log.info(`Experimentation Audit completed in ${formattedElapsed} seconds for ${baseURL}`);
+  return auditData;
 }
+
+export default new AuditBuilder()
+  .withRunner((finalURL, context) => experimentationAuditRunner(finalURL, context))
```

Comment:
you don't even need an anonymous function, you can do directly:
```suggestion
  .withRunner(experimentationAuditRunner)
```

Reply:
Have implemented this.